config/resource.go

```diff
@@ -136,6 +137,13 @@ func (config *Component) Validate(path string) ([]string, error) {
 	if config.Name == "" {
 		return nil, utils.NewConfigValidationFieldRequiredError(path, "name")
 	}
+	matched, err := regexp.MatchString(`^\w+$`, config.Name)
```

Comment:
We should just do a simple character check here. I think underscore is okay but you can only start with a letter.